es717x/src/main/scala/tech/beshu/ror/es/actions/wrappers/_cat/TransportRorWrappedCatAction.scala

```diff
@@ -20,24 +20,28 @@ import org.elasticsearch.action.ActionListener
 import org.elasticsearch.action.support.{ActionFilters, HandledTransportAction}
 import org.elasticsearch.common.inject.Inject
 import org.elasticsearch.tasks.Task
+import org.elasticsearch.threadpool.ThreadPool
 import org.elasticsearch.transport.TransportService
 
 import scala.annotation.nowarn
 
 class TransportRorWrappedCatAction(transportService: TransportService,
                                    actionFilters: ActionFilters,
+                                   threadPool: ThreadPool,
                                    @nowarn("cat=unused") constructorDiscriminator: Unit)
   extends HandledTransportAction[RorWrappedCatRequest, RorWrappedCatResponse](
     RorWrappedCatActionType.name, transportService, actionFilters, RorWrappedCatActionType.exceptionReader
   ) {
 
   @Inject
   def this(transportService: TransportService,
-           actionFilters: ActionFilters) =
-    this(transportService, actionFilters, ())
+           actionFilters: ActionFilters,
+           threadPool: ThreadPool) =
+    this(transportService, actionFilters, threadPool, ())
 
   override def doExecute(task: Task, request: RorWrappedCatRequest,
                          listener: ActionListener[RorWrappedCatResponse]): Unit = {
+    threadPool.getThreadContext.addResponseHeader("x-elastic-product", "elasticsearch")
```

Comment:
Why does this apply only to the es717x version? Is the header present in the further versions?

Reply:
sorry, I forgot to clean it up